File: julia-code/computational-implementation.jl
using Pkg; Pkg.activate("."); Pkg.instantiate();
using BayesFlux, Flux
using Distributions

net = Chain(Dense(5, 1))
net = Chain(Dense(5, 5, relu), Dense(5, 1))
net = Chain(RNN(5, 5), Dense(5, 5, sigmoid), Dense(5, 1))
net = Chain(LSTM(5, 5), LSTM(5, 3), Dense(3, 1))

nc = destruct(net)
prior = GaussianPrior(nc, 0.5f0)

like = FeedforwardNormal(nc, Gamma(2.0, 0.5))

init = InitialiseAllSame(Normal(0.0f0, 0.5f0), like, prior)

File: r-code/computational-implementation.R
library(BayesFluxR)
BayesFluxR_setup(installJulia = TRUE, env_path = ".", seed = 123456)

net <- Chain(Dense(5, 1))
net <- Chain(Dense(5, 5, "relu"), Dense(5, 1))
net <- Chain(RNN(5, 5), Dense(5, 5, "sigmoid"), Dense(5, 1))
net <- Chain(LSTM(5, 5), LSTM(5, 3), Dense(3, 1))

prior <- prior.gaussian(net, 0.5)
like <- likelihood.feedforward_normal(net, Gamma(2.0, 0.5))
init <- initialise.allsame(Normal(0, 0.5), like, prior)
